ion/integration_test.go

```diff
@@ -0,0 +1,809 @@
+	"nonIVMNoOps.ion",
+	"sexps.ion",
+	"stringUtf8.ion",
+	"strings.ion",
+	"structsFieldsDiffOrder.ion",
+	"structsFieldsRepeatedNames.ion",
+	"systemSymbols.ion",
+	"systemSymbolsAsAnnotations.ion",
+	"timestampSuperfluousOffset.10n",
+	"timestamps.ion",
+	"timestampsLargeFractionalPrecision.ion",
+}
+
+var nonEquivsSkipList = []string{
+	"bools.ion",
+	"decimals.ion",
+	"documents.ion",
+	"floats.ion",
+	"floatsVsDecimals.ion",
+	"localSymbolTableWithAnnotations.ion",
+	"structs.ion",
+	"symbolTables.ion",
+	"symbolTablesUnknownText.ion",
+	"symbols.ion",
+	"timestamps.ion",
+}
+
+func TestBinaryRoundTrip(t *testing.T) {
+	readFilesAndTest(t, goodPath, binaryRoundTripSkipList, func(t *testing.T, path string) {
+		binaryRoundTrip(t, path)
+	})
+}
+
+func TestTextRoundTrip(t *testing.T) {
+	readFilesAndTest(t, goodPath, textRoundTripSkipList, func(t *testing.T, path string) {
+		textRoundTrip(t, path)
+	})
+}
+
+func TestLoadBad(t *testing.T) {
+	readFilesAndTest(t, badPath, malformedIonsSkipList, func(t *testing.T, path string) {
+		testLoadBad(t, path)
+	})
+}
+
+func TestEquivalency(t *testing.T) {
+	readFilesAndTest(t, equivsPath, equivsSkipList, func(t *testing.T, path string) {
+		testEquivalency(t, path, true)
+	})
+}
+
+func TestNonEquivalency(t *testing.T) {
+	readFilesAndTest(t, nonEquivsPath, nonEquivsSkipList, func(t *testing.T, path string) {
+		testEquivalency(t, path, false)
+	})
+}
+
+func binaryRoundTrip(t *testing.T, fp string) {
+	b := loadFile(t, fp)
+
+	// Make a binary writer from the file
+	r := NewReaderBytes(b)
+	buf := bytes.Buffer{}
+	bw := NewBinaryWriter(&buf)
+	writeToWriterFromReader(t, r, bw)
+	bw.Finish()
+
+	// Make a text writer from the binary writer
+	r = NewReaderBytes(buf.Bytes())
+	str := strings.Builder{}
+	tw := NewTextWriter(&str)
+	writeToWriterFromReader(t, r, tw)
+	tw.Finish()
+
+	// Make another binary writer using the text writer
+	r = NewReaderStr(str.String())
+	buf2 := bytes.Buffer{}
+	bw2 := NewBinaryWriter(&buf2)
+	writeToWriterFromReader(t, r, bw2)
+	bw2.Finish()
+
+	// Compare the 2 binary writers
+	if !reflect.DeepEqual(bw, bw2) {
+		t.Errorf("Round trip test failed on: " + fp)
+	}
+}
+
+func textRoundTrip(t *testing.T, fp string) {
+	b := loadFile(t, fp)
+
+	// Make a text writer from the file
+	r := NewReaderBytes(b)
+	str := strings.Builder{}
+	tw := NewTextWriter(&str)
+	writeToWriterFromReader(t, r, tw)
+	tw.Finish()
+
+	// Make a binary writer from the text writer
+	r = NewReaderStr(str.String())
+	buf := bytes.Buffer{}
+	bw := NewBinaryWriter(&buf)
+	writeToWriterFromReader(t, r, bw)
+	bw.Finish()
+
+	// Make another text writer using the binary writer
+	r = NewReaderBytes(buf.Bytes())
+	str2 := strings.Builder{}
+	tw2 := NewTextWriter(&str2)
+	writeToWriterFromReader(t, r, tw2)
+	tw2.Finish()
+
+	//compare the 2 text writers
+	if !reflect.DeepEqual(tw, tw2) {
+		t.Errorf("Round trip test failed on: " + fp)
+	}
+}
+
+func testLoadBad(t *testing.T, fp string) {
```

Comment:
Could you add a one-line comment to each of the `test*` helper functions explaining their intent?

Reply:
Done.